src/events/stickyMessageHandler.ts

```diff
@@ -0,0 +1,42 @@
+import { Events } from 'discord.js'
+
+import { StickyMessage } from '@prisma/client'
+
+import { isChannelLock } from '../features/stickyMessage/channelLock.js'
+import {
+  STICKY_CACHE_PREFIX,
+  isNeedToUpdateMessage,
+  pushMessageToBottom,
+} from '../features/stickyMessage/index.js'
+import { incCounter } from '../features/stickyMessage/messageCounter.js'
+import { defineEventHandler } from '../types/defineEventHandler.js'
+import { getCache } from '../utils/cache.js'
+
+export default defineEventHandler({
+  eventName: Events.MessageCreate,
+  once: false,
+  execute: async (_botContext, message) => {
+    const stickyMessage = getCache(
+      `${STICKY_CACHE_PREFIX}-${message.channelId}`,
+    ) as StickyMessage
+
+    // if channel not has any sticky message -> do nothing
+    if (!stickyMessage) {
+      return
+    }
+
+    // if message doesn't need to update -> increase message counter
+    if (!isNeedToUpdateMessage(message.channelId)) {
+      incCounter(message.channelId)
+      return
+    }
+
+    // if message is already updated -> do nothing
```

Comment:
This might be nitpick but did you mean **updating**, its lock when message is being update right?

Reply:
yes, it locked when the bot updating the message to bottom of channel to prevent the race condition

Reply:
do I need to change anything in this statement?